### DISCUSSION TITLE
Only allow Golang manager for specific repository

### DISCUSSION THREAD

**Discussion by dirsigler** [renovatebot/renovate#19031] [Request Help] [answered]
Hello,

we have Renovatebot configured to autodiscover all Helm and Terraform projects. These projects are all maintained by the Ops Team.

```yaml
config: |
        {
[...]
          "autodiscover": true,
          "enabledManagers": ["helmv3", "terraform"],
[...]
        }
```

We would like to add now the Golang manager for only specific repositories while keeping the auto-discovery of Helm and Terraform projects.

How I can do it?

**Comment by rarkins** [comment, marked as answer]
If you want to do it all centrally, you can selectively adjust individual repo config like so:

```
config: |
        {
[...]
          "autodiscover": true,
          "enabledManagers": ["helmv3", "terraform"],
          "repositories: [{
            "repository": "foo/bar",
            "enabledManagers": ["gomod", "helmv3", "terraform"],
          }, {
            "repository": "another/one",
            "enabledManagers": ["gomod", "helmv3", "terraform"],
          }
[...]
        }
```

**Reply by dirsigler** [reply]
Hey,
thank you very much @rarkins for the fast answer.
I tested this out now with one repository. As it seems Renovatebot finds the repository successfully but complains for the lack of `go.mod` even tho it is there in the root of the project alongside a `go.sum`.

Is there anything that I am missing?

**Reply by rarkins** [reply]
I realized I had a typo in the above btw, it should be `gomod` and not `golang`. Did you already figure that part out?

If so, could you share the log messages you are referring to? Normally it doesn't complain about lack of file, it simply just moves past it

**Reply by dirsigler** [reply]
ohhh no, I did not discover this typo.

**Comment by dirsigler** [comment]
So as rarkins correctly suggested, this is how my config part has to look like now:

```
config: |
        {
[...]
          "autodiscover": true,
          "enabledManagers": ["helmv3", "terraform"],
          "repositories": [{
            "repository": "foo/bar",
            "enabledManagers": ["gomod"]
          }],
[...]
}
```

Thank you @rarkins !
